lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_emulator_helper.rb

```diff
@@ -0,0 +1,198 @@
+        end
+
+        # @return [Array<Fastlane::Helper::AdbDevice>] List of currently booted emulators
+        #
+        def running_emulators
+          helper = Fastlane::Helper::AdbHelper.new(adb_path: @tools.adb)
+          helper.load_all_devices.select { |device| device.serial.include?('emulator') }
+        end
+
+        def find_serial(avd_name:)
+          running_emulators.find do |candidate|
+            command = [@tools.adb, '-s', candidate.serial, 'emu', 'avd', 'name']
+            UI.command(command.shelljoin)
+            candidate_name = Actions.sh(*command, log: false).split("\n").first.chomp
+            candidate_name == avd_name
+          end&.serial
+        end
+
+        # Trigger a shutdown for all running emulators, and wait until there is no more emulator running.
+        #
+        # @param [Array<String>] serials List of emulator serials to shut down. Will shut down all of them if `nil`.
+        #
+        def shut_down_emulators!(serials: nil)
+          UI.message("Shutting down #{serials || 'all'} emulator(s)...")
+
+          emulators_list = running_emulators.map(&:serial)
+          # Get the intersection of the set of running emulators with the ones we want to shut down
+          emulators_list &= serials unless serials.nil?
+          emulators_list.each do |e|
+            Actions.sh(@tools.adb, '-s', e, 'emu', 'kill') { |_| } # ignore error if no emulator with specified serial is running
+
+            # NOTE: Alternative way of shutting down emulator would be to call the following command instead, which shuts down the emulator more gracefully:
+            # `adb -s #{e} shell reboot -p` # In case you're wondering, `-p` is for "power-off"
+            # But this alternate command:
+            #  - Requires that `-no-snapshot` was used on boot (to avoid being prompted to save current state on shutdown)
+            #  - Disconnects the emulator from `adb` (and thus disappear from `adb devices -l`) for a short amount of time,
+            #    before reconnecting to it but in an `offline` state, until `emulator` finally completely quits and it disappears
+            #    again (for good) from `adb devices --list`.
+            # This means that so if we used alternative, we couldn't really retry_loop until emulator disappears from `running_emulators` to detect
+            # that the shutdown was really complete, as we might as well accidentally detect the intermediate disconnect instead.
+          end
+
+          # Wait until all emulators are killed
+          retry_loop(time_between_retries: SHUTDOWN_WAIT, timeout: SHUTDOWN_TIMEOUT, description: 'waiting for devices to shutdown') do
+            (emulators_list & running_emulators.map(&:serial)).empty?
+          end
+
+          UI.success('All emulators are now shut down.')
+        end
+
+        # Find the system-images package for the provided API, with Google APIs, and matching the current platform/architecture this lane is called from.
```

Comment:
Nitpick, to make it consistent with the usage [above](https://github.com/wordpress-mobile/release-toolkit/pull/409/files#diff-d271fcb6de300236feaec1c6b889596caf09c69201f3f8d89a692cd0496dd8ddR17)

```suggestion
        # Find the system-images package for the provided `api`, with Google APIs, and matching the current platform/architecture this lane is called from.
```

Personally, I prefer using `api` over API because it makes it clearer that we are referring to the method paratmer.